src/main/java/org/jboss/xavier/integrations/jpa/service/AnalysisService.java

```diff
@@ -123,4 +125,17 @@ public void updateStatus(String status, Long id) {
     {
         return analysisRepository.getAdministrationMetrics(fromDate, toDate);
     }
+
+    public void markAsFailedIfNotCreated(Long id) {
+        AnalysisModel analysisModel = findByIdAndStatusIgnoreCaseNot(id, STATUS.CREATED.toString());
+        if (analysisModel != null) {
+            analysisModel.setStatus(STATUS.FAILED.toString());
+            analysisModel.setLastUpdate(new Date());
+            analysisRepository.save(analysisModel);
+        }
+    }
+
+    private AnalysisModel findByIdAndStatusIgnoreCaseNot(Long id, String status) {
```

Comment:
Worth adding a unit test for this new method in `AnalysisServiceTest`

Reply:
+1